v2/bundle/jwtbundle/bundle.go

```diff
@@ -1,73 +1,164 @@
 package jwtbundle
 
 import (
+	"bytes"
 	"crypto"
+	"encoding/json"
+	"errors"
+	"fmt"
 	"io"
+	"io/ioutil"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"gopkg.in/square/go-jose.v2"
 )
 
 // Bundle is a collection of trusted JWT public keys for a trust domain.
 type Bundle struct {
+	mtx            sync.RWMutex
+	trustDomain    spiffeid.TrustDomain
+	jwtSigningKeys map[string]crypto.PublicKey
 }
 
 // New creates a new bundle.
 func New(trustDomain spiffeid.TrustDomain) *Bundle {
-	panic("not implemented")
+	return &Bundle{
+		mtx:            sync.RWMutex{},
```

Comment:
Initializing with a zero value is unnecessary. The field will already be initialized with a zero value.